Add option to use Adam 8bit to reduce VRAM usage

flesnuk opened the pull request:
I also added the bitsandbytes_windows folder for simplyfing the installation in Windows workspace (by providing the cp instuctions in the README). bitandbytes is not supported natively on Windows, but there are prebuilt dll binaries from here https://github.com/DeXtmL/bitsandbytes-win-prebuilt 

But it can be removed if you don't like adding binaries/patch for windows to this project and add a link to the repo with prebuilt binaries.

jumon reviewed (changes requested):
Thank you for your recent valuable contributions! The addition of support for 8bit optimizers is fantastic and will be particularly beneficial for those with limited computation resources.

I would like to make this repository friendly for Windows users, but also would like to keep it simple. Would you be able to remove the `bitsandbytes_windows` folder and instead provide some instructions/links for Windows users in the README file?

flesnuk commented:
Hi, that is understandable, I removed the folder and added instructions for anyone to setup it easily.
I forgot to note that I changed the requirements.txt and pinned the versions to avoid issues in the future with Whisper package updating functions and breaking the project when running. Only downside is having to update the requirements from time to time, but I think it is better this way.

jumon reviewed (approved):
Apologies for the delayed response. I have been occupied with some personal matters.

I have reviewed the changes and they look excellent. It is wise to specify the versions of the libraries. Thank you! I made a few modifications to the README.
